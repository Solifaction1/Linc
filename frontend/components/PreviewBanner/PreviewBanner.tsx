import React from "react";
import Router from "next/router.js";
import Button from "@components/Button/Button";

import styles from "./previewBanner.module.scss";

const PreviewBanner: React.FC = () => {
  return (
    <div className={styles.previewBanner}>
      <span>You are currently viewing in Preview Mode. </span>
      <Button
        label="Turn off preview mode"
        size="sm"
        handleClick={() => exitPreviewMode()}
      />
    </div>
  );
};

async function exitPreviewMode() {
  const response = await fetch("/api/exit-preview");
  if (response) {
    Router.reload();
  }
}

export default PreviewBanner;
