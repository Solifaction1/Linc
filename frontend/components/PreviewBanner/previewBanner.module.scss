@import 'styles/vars';
@import 'styles/utilities';

.previewBanner {
	@extend %absolute-banner;
	z-index: 2;

	width: 100%;
	padding: 1rem;

	background-color: $color-beige;
	color: $color-white;
	box-shadow: $box-shadow;

	font-family: $font-family-title;
}
