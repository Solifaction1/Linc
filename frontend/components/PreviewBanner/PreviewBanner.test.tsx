import React from 'react'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import PreviewBanner from './PreviewBanner'

describe('PreviewBanner', () => {
	it('renders a PreviewBanner', () => {
		const { container, getByText } = render(<PreviewBanner />)

		expect(container).toMatchSnapshot()

		expect(container.getElementsByClassName('previewBanner')[0]).toBeVisible()

		expect(
			getByText('You are currently viewing in Preview Mode.', {
				selector: 'span',
			})
		).toBeVisible()

		expect(
			container.getElementsByClassName('button false primary sm')[0]
		).toBeVisible()
		expect(
			getByText('Turn off preview mode', {
				selector: 'p',
			})
		).toBeVisible()
	})
})
