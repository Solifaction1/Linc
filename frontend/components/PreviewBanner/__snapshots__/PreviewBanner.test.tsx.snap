// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewBanner renders a PreviewBanner 1`] = `
<div>
  <div
    class="previewBanner"
  >
    <span>
      You are currently viewing in Preview Mode. 
    </span>
    <button
      class="button false primary sm false  beige"
      type="button"
    >
      <p
        class="label"
      >
        Turn off preview mode
      </p>
    </button>
  </div>
</div>
`;
